### PR TITLE
create-gcloud-instance (pre-requisite)

### PR DESCRIPTION
**Pull request by iMichka**
These scripts are needed to setup the self hosted runners

### CODE REVIEW

File: .github/workflows/test-create-gcloud-instance.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Test create gcloud install
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
Comment (MikeMcQuaid) [inline .github/workflows/test-create-gcloud-instance.yml:6]: Add something like https://github.com/Homebrew/actions/blob/560e336611c852f0ab459a16e3041001fc39ec7b/.github/workflows/cancel-previous-runs.yml#L5-L9 so it's tested in PRs like this.

File: .github/workflows/test-create-gcloud-instance.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+      - name: Install shellcheck
+        run: brew install shellcheck
+      - name: Check scripts
+        run: |
+          shellcheck create-gcloud-instance/config_runner.sh
+          shellcheck create-gcloud-instance/install_runner.sh
```
Comment (MikeMcQuaid) [inline .github/workflows/test-create-gcloud-instance.yml:20]:
```suggestion
          shellcheck --enable=all create-gcloud-instance/*.sh
```

**Comment by iMichka** [comment]
```
2021-05-07T11:05:51.0433638Z In create-gcloud-instance/config_runner.sh line 6:
2021-05-07T11:05:51.0434724Z   --url "https://github.com/${REPO_NAME}" \
2021-05-07T11:05:51.0435705Z                             ^----------^ SC2154: REPO_NAME is referenced but not assigned.
```

I'm wondering how to handle env variables to make the style check pass.

**Comment by MikeMcQuaid** [comment]
> I'm wondering how to handle env variables to make the style check pass.

What sets these values?

**Comment by iMichka** [comment]
For example https://github.com/Homebrew/actions/pull/171/files#diff-3a807529c1045736f2955ba4a30f8b8c3b99ffe285676a815b979c22506e2e75R63

So the script that calls this script.

**Comment by MikeMcQuaid** [comment]
@iMichka Cool. In that case you can add a shellcheck ignore line and a comment above which explains that these variables are set from the script environment.

**Comment by iMichka** [comment]
All done

File: .github/workflows/test-create-gcloud-instance.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+name: Test create gcloud install
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    paths: 
+      - '**create-gcloud-instance**'
```
Comment (MikeMcQuaid) [inline .github/workflows/test-create-gcloud-instance.yml:8]:
```suggestion
      - 'create-gcloud-instance/'
```

If it works

File: create-gcloud-instance/config_runner.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# shellcheck disable=SC2154
+# SC2154: REPO_NAME/VM_TOKEN/RUNNER_NAME is referenced but not assigned: 
+# These variables are set from the script environment.
+set -eo pipefail
+
+cd /home/actions/actions-runner || exit
+./config.sh \
+  --url "https://github.com/${REPO_NAME}" \
```
Comment (MikeMcQuaid) [inline create-gcloud-instance/config_runner.sh:9]:
```suggestion
set -eo pipefail

cd /home/actions/actions-runner || exit

# shellcheck disable=SC2154
# SC2154: REPO_NAME/VM_TOKEN/RUNNER_NAME is referenced but not assigned: 
# These variables are set from the script environment.
./config.sh \
  --url "https://github.com/${REPO_NAME}" \
```

**Review by MikeMcQuaid — APPROVED** [review]
Looks great, thanks @iMichka!

**Pull request merged**